lua/wire/server/sents_registry.lua

```diff
@@ -33,43 +38,56 @@
+
+-- Tbh you can use this as you want, but make sure not to override existing entries, to not break propSpawn (or do it, if you know what you're doing).
+
+-- WARNING: Obviously, you shouldn't add any sents by WireLib.SentSpawn.Register, if you don't want them to be spawned by propcore's sentSpawn function.
+--		(Or blacklist them beforehand)
+
 local GetOwner = WireLib.GetOwner
-local SentSpawn = E2Lib.SentSpawn
+local SentSpawn = WireLib.SentSpawn
 if not SentSpawn then
 	SentSpawn = {}
-	E2Lib.SentSpawn = SentSpawn
+	WireLib.SentSpawn = SentSpawn
 end
 
--- Registers new class to be able to be spawned using sentSpawn, and appends it to 'wire_spawnable_ents_whitelist' list.
--- Even tho it can be used outside of this file, I would recommend appending your entities here. (To keep everything tidy.)
--- This is made mainly to support thirdparty addons.
+-- Registers new class to be able to be spawned using sentSpawn, and appends it to 'wire_spawnable_ents_registry' list.
+-- Even tho it's global function, I would recommend appending your entities here in this file. (To keep everything tidy and organized).
+-- It've been made global to support thirdparty addons.
 ---@param class string
 ---@param data table
-function SentSpawn.WhitelistAdd(class, data)
+function SentSpawn.Register(class, data)
 	if TypeID(class) ~= TYPE_STRING then ErrorNoHaltWithStack("Class type must be TYPE_STRING!") return end
 	if TypeID(data) ~= TYPE_TABLE then ErrorNoHaltWithStack("Data type must be TYPE_TABLE") return end
 
-	list.Set("wire_spawnable_ents_whitelist", class, data)
+	list.Set("wire_spawnable_ents_registry", class, data)
 end
 
--- Deletes registered sent class and it's data from 'wire_spawnable_ents_whitelist' list. (So it can no longer be spawned with sentSpawn.)
--- Even tho it can be used outside of this file, I would recommend removing your entities here. (To keep everything tidy.)
--- This is made mainly to support thirdparty addons.
+-- Deletes registered sent class and it's data from 'wire_spawnable_ents_registry' list.
+-- Even tho it can be used as a way to blacklist an entity from being spawned by propcore's sentSpawn function, better use use Expression2_CanSpawnSent hook, or statically
+-- blacklist an entity in lua/entities/gmod_wire_expression2/core/custom/prop.lua:201 (local blacklistedSents table).
+-- It've been made global to support thirdparty addons.
 ---@param class string
-function SentSpawn.WhitelistRemove(class)
-	if not list.HasEntry("wire_spawnable_ents_whitelist", class) then ErrorNoHaltWithStack("Trying to remove entity that is not registered!") return end
+function SentSpawn.Unregister(class)
+	if not list.HasEntry("wire_spawnable_ents_registry", class) then ErrorNoHaltWithStack("Trying to remove entity that is not registered!") return end
 
-	local whitelist = list.GetForEdit("wire_spawnable_ents_whitelist")
+	local whitelist = list.GetForEdit("wire_spawnable_ents_registry")
 	table.remove(whitelist, class)
```

Comment:
Shouldn't this be `whitelist[class]=nil`? I don't think table.remove works on string indexed tables.

Reply:
Yeah you're right, my bad